MyToolsYourToolsBackend/MyToolsYourToolsBackend.API/Controllers/OpinionsController.cs

```diff
@@ -0,0 +1,48 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Cors;
+using MyToolsYourToolsBackend.Application.Services;
+using MyToolsYourToolsBackend.Application.Dtos;
+
+namespace MyToolsYourToolsBackend.API.Controllers
+{
+    [Route("api")]
+    [ApiController]
+    [EnableCors("MyToolsYourTools")]
+    public class OpinionsController : ControllerBase
+    {
+
+        private IOpinionService _opinionService;
+        private IUserService _userService;
+
+        public OpinionsController(IOpinionService opinionService, IUserService userService)
+        {
+            _opinionService = opinionService;
+            _userService = userService;
+        }
+
+        [HttpGet("opinions")]
+        public IActionResult GetOpinions()
+        {
+            return Ok(_opinionService.GetAllOpinions());
+        }
+
+        [HttpPost("{ratedUserId}/{ratingUserId}/opinions")]
+        public IActionResult AddOpinion([FromBody]OpinionForCreationDto opinionFromBody, Guid ratedUserId, Guid ratingUserId)
+        {
+            if (!_userService.CheckIfUserExists(ratedUserId))
+            {
+                return NotFound();
+            }
+
+            var opinionToReturn = _opinionService.AddOpinion(opinionFromBody, ratedUserId, ratingUserId);
+
+            return Created(nameof(GetOpinions), opinionToReturn);
+        }
+
+    }
+}
```

Comment:
Ogólnie do kontrolera dodałbym funkcję typu GetUserOpinions(userId), żeby za każdym razem nie ściągać na front wszystkich opinii i filtrować ich na froncie, tylko przefiltrowane już zwrócić z backendu, ale na razie możemy to zostawić, bo działa dobrze